napari/utils/action_manager.py

```diff
@@ -0,0 +1,255 @@
+            tooltips.
+
+        """
+        # if name in self._actions:
+        #    import warnings
+
+        #    warnings.warn(
+        #        f'Warning, action {name} already exists, make sure '
+        #        'you are not overwriting an existing action',
+        #        stacklevel=2,
+        #    )
+        assert description is not None
+        self._actions[name] = Action(command, description, keymapprovider)
+        self._update_shortcut_bindings(name)
+        self._update_gui_elements(name)
+
+    def _update_gui_elements(self, name):
+        """
+        Update the description and shortcuts of all the (known) gui elements.
+        """
+        if name not in self._actions:
+            return
+        buttons = self._buttons.get(name, [])
+        desc = self._actions[name].description
+
+        # update buttons with shortcut and description
+        if name in self._shortcuts:
+            sht = self._shortcuts[name]
+            sht_str = f' ({Shortcut(sht).platform})'
+        else:
+            sht = ''
+            sht_str = ''
+
+        for button in buttons:
+            button.setToolTip(desc + sht_str)
+            action = self._actions[name]
+
+            try:
+                # not sure how to check whether things are connected already
+                button.clicked.disconnect(action.callable(self.context))
+            except Exception:
+                pass
+            button.clicked.connect(action.callable(self.context))
+        if name in self._qactions:
+            action = self._actions[name]
+            qaction = self._qactions[name]
+            # We can't set the shortcut yet, or this will later break the layers
+            # when we use the action manager for those. Setting the key sequence
+            # there will _also_ bind the shortcuts but at the window level which
+            # has a higher priority. SO this conflict with KeymapProvider
+            #
+            # qaction.setShortcut(
+            #     QKeySequence(sht.replace('-', '+').replace('Control', 'Ctrl'))
+            # )
+            menu_name = name
+            if ' ' not in menu_name:
+                components = [
+                    word.capitalize() for word in menu_name.split('_')
+                ]
+                menu_name = ' '.join(components)
+            qaction.setText(menu_name)
+            qaction.setStatusTip(action.description)
+
+    def _update_shortcut_bindings(self, name):
+        """
+        Update the key mappable for given action name
+        to trigger the action within the given context and
+        """
+        if name not in self._actions:
+            return
+        action = self._actions[name]
+        if name not in self._shortcuts:
+            return
+        sht = self._shortcuts.get(name)
+        keymapprovider = action.keymapprovider
+        if hasattr(keymapprovider, 'bind_key'):
+            keymapprovider.bind_key(sht, overwrite=True)(action.command)
```

Comment:
I now need `overwrite=True`,  because when we register shortcuts  in instances, in tests they would trigger errors on tests each time we call `make_napari_viewer()` fixture. 

I've done this to try to addresses the question like [this one](https://github.com/napari/napari/pull/2495#discussion_r613609511), we can try to tackle later about separating the action registration from the shortcuts.